week0003/daudau/php/daudau.php

```diff
@@ -0,0 +1,192 @@
+        else
+            $this->_accounts[strtolower( $account->getName() )] = $account;
+        return true;
+    }
+
+    public function save($account) {
+        if ( $this->_accounts[strtolower($account->getName())] ) {
+            $this->_accounts[strtolower($account->getName())] = $account;
+            return true;
+        }
+        return false;
+    }
+
+    public function run() {
+        while ($command = dd_read()) {
+
+            if ($command == 'dump') {
+                print_r($this->_accounts);
+                continue;
+            }
+
+            if ($command = $this->parseCommand($command)) {
+                $name = strtolower($command['accountName']); //this name use as key of bank per account.
+                
+                switch ($command['action']) {
+                    case self::ACTION_ADD_CLIENT:
+                        $account =  new Account($command['accountName'], $command['amount']);
+                        if ($this->add($account))
+                            dd_linef('True');
+                        else
+                            dd_error('False');
+                        break;
+
+                    case self::ACTION_DEPOSITE_CLIENT:
+                        if ($acc = $this->findAccountByName($name)) {
+                            if ($acc->deposite($command['amount'])) {
+                                $acc->save();
+                                dd_linef('True');
+                            } else
+                                dd_error('False');
+                        } else
+                            dd_error('False');
+                        break;
+
+                    case self::ACTION_WITHDRAW_CLIENT:
+                        if ($acc = $this->findAccountByName($name)) {
+                            if ($acc->withdraw($command['amount'])) {
+                                $acc->save();
+                                dd_linef('True');
+                            } else
+                                dd_error('False');
+                        } else
+                            dd_error('False');
+                        break;
+                    
+                    default:
+                        dd_error('Invalid');
+                        break;
+                }
+            }
+        }
+    }
+
+    public function findAccountByName($name) {
+        if ($this->_accounts[$name])
+            return $this->_accounts[$name];
+        return false;
+    }
+
+    private function parseCommand($string) {
+        $string = preg_split('/\s+/', $string);
+        $errors = array();
+        
+        //just usefull when all commands have 3 arguments
+        if (count($string) != 3) {
+            dd_linef("Wrong structure command!");
+            return false;
+        }
+
+        $command = array();
+        $command['action'] = $string[0];
+        $command['accountName'] = $string[1];
+        $command['amount'] = $string[2];
```

Comment:
in my case, if amount <= 0 it still printr false like command not complete.
I need valid if $command['amount'] can be convert into a double number in line laters.